File: Lab3/src/Controller/Controller.java
package Controller;
import Repository.*;
import Model.*;
import Statement.*;
import Exception.*;

public class Controller
{
    private IPrgStateRepo ipsr;
    public Controller(IPrgStateRepo ip)
    {
        ipsr = ip;
    }
    public void executeOneStep()
    {
        PrgState ps = ipsr.getCurrentProgram();
        IExecStack<Statement> ex = ps.getExecStack();
        if(! ex.isEmpty())
        {
            Statement instr = ex.pop();
            try
            {
                instr.execute(ps);
            }
            catch(InterpretorException e)
            {
                System.out.println(e.getMessage());
            }
            System.out.println(ps);
        }
        System.out.println("----------------------------------------------");
    }
    public void executeAll()
    {
        PrgState ps = ipsr.getCurrentProgram();
        IExecStack<Statement> es = ps.getExecStack();
        while(! es.isEmpty())
        {
            try
            {
                executeOneStep();
                ipsr.logPrgStateExec();
            }
            catch(InterpretorException e)
            {
                System.out.println(e.getMessage());
            }
        }
        System.out.println("----------------------------------------------");
    }
}

File: MAP_Lab7/MAP_Lab7/ControllerFile/Controller.cs
using MAP_Lab7.Statements;
using MAP_Lab7.Repository;
using MAP_Lab7.Model;
using MAP_Lab7.Exceptions;
using System;

namespace MAP_Lab7.ControllerFile
{
    public class Controller
    {
        private IPrgStateRepo repo;

        public Controller(IPrgStateRepo r)
        {
            repo = r;
        }
        
        public void executeOneStep()
        {
            PrgState ps = repo.GetCurrentProgram();
            IExeStack<Statement> ex = ps.ExeStack;
            if (!ex.IsEmpty())
            {
                Statement stmt = ex.PopS();
                stmt.Execute(ps);
            }
            Console.WriteLine("----------------------------------------------\n\n");
        }

        public void executeAll()
        {
            PrgState ps = repo.GetCurrentProgram();
            IExeStack<Statement> es = ps.ExeStack;
            while (!es.IsEmpty())
            {
                try
                {
                    executeOneStep();
                    repo.logPrgStateExec();
                }
                catch(FileException e)
                {
                    Console.WriteLine(e.Message);
                } 
            }
            Console.WriteLine("----------------------------------------------\n\n");
        }
    }
}